presto-hive/src/main/java/com/facebook/presto/hive/OrcFileWriterFactory.java

```diff
@@ -105,23 +108,30 @@
      * A boolean value, stored in the SerDe properties as a string, indicating
      * that the flat map writer for this table should be enabled.
      */
-    private static final String ORC_FLAT_MAP_WRITER_ENABLED_KEY = "orc.flatten.map";
+    static final String ORC_FLAT_MAP_WRITER_ENABLED_KEY = "orc.flatten.map";
+
+    /**
+     * An integer value stored in the SerDe properties as a string, and indicating
+     * the max allowed number of keys in a flat map column.
+     */
+    static final String ORC_FLAT_MAP_KEY_LIMIT_KEY = "orc.map.flat.max.keys";
 
     /**
      * A comma separated list of column numbers, stored in the SerDe properties,
      * indicating which columns should use flat map writer.
      */
-    private static final String ORC_FLAT_MAP_COLUMN_NUMBERS_KEY = "orc.map.flat.cols";
-    private static final Splitter FLAT_MAP_COLUMN_NUMBERS_SPLITTER = Splitter.on(',').omitEmptyStrings().trimResults();
+    static final String ORC_FLAT_MAP_COLUMN_NUMBERS_KEY = "orc.map.flat.cols";
 
     /**
      * A boolean value, stored in the SerDe properties as a string, indicating
      * that the flat map writer should generate map column statistics instead of
      * compound column statistics.
      */
-    private static final String ORC_MAP_STATISTICS_KEY = "orc.map.statistics";
+    static final String ORC_MAP_STATISTICS_KEY = "orc.map.statistics";
```

Comment:
why is the word flat missing in this property ? If no one uses it, will recommend adding it to indiciate it is only for flat maps.

Reply:
According to Jake the idea is that we should be able to use it to for regular maps, at least at some point. That's why it is not tied to flat maps specifically.

Reply:
Isn't that confusing the existing users while we know that in the future this will certainly not be a priority :)

Reply:
Yeah, it makes sense. We can give it a better name in Velox or Alpha 👍